os/pm/pm_procfs.c

```diff
@@ -199,155 +200,77 @@ const struct procfs_operations power_procfsoperations = {
 
 static int power_find_dirref(FAR const char *relpath, FAR struct power_dir_s *dir)
 {
-	uint16_t len;
-	FAR char *str;
-	int i;
-
-	/* Skip the "power/domains" portion of relpath. We accept it only now */
-	len = strlen(POWER_SUBDIRPATH_DOMAINS);
-	if (strncmp(relpath, POWER_SUBDIRPATH_DOMAINS, len) != 0) {
-		fdbg("Invalid Path : Failed to find path %s \n", relpath);
-		return -ENOENT;
-	}
-	relpath += len;
-
-	if (relpath[0] == '/') {
-		relpath++;
-	}
-
-	if (relpath[0] == '\0') {
-		/* Requesting directory listing of registered device drivers for PM */
-		dir->base.level = 1;
-		dir->base.index = 0;
-		dir->base.nentries = CONFIG_PM_NDOMAINS;
-		/* Search PM registry to the number of registered device driver */
-		return OK;
-	}
-
-	str = NULL;
-
-	if (!str) {
-		fdbg("ERROR: Invalid path \"%s\"\n", relpath);
-		return -ENOENT;
-	}
-
-	len = str - relpath;
-
-	relpath += len;
-
-	if (relpath[0] == '/') {
-		relpath++;
-	}
-
-	if (relpath[0] == '\0') {
-		/* Requesting directory listing of a specific device driver or entry */
-		dir->base.level = 2;
-		dir->base.index = 0;
-		dir->base.nentries = g_power_direntrycount;
-		dir->direntry = 0;
-		return OK;
-	}
-
-	/* This is 3 level. Only files exist only in this level. */
-	dir->base.level = 3;
-	dir->base.nentries = 1;
-	dir->base.index = 0;
-	dir->direntry = 0;
-	for (i = 0; i < g_power_direntrycount; i++) {
-		if (strcmp(relpath, g_power_direntry[i].name) == 0) {
-			dir->direntry = i;
+	char *str;
+	int domain_id;
+	/* Function to check path is starting with given name */
+	int checkStart(char *name, bool isDirectory) {
```

Comment:
Could you let me know adv of this? I mean, why do you define this function inside of other function?

Reply:
We were using same set of code repetitively to compare string path with folder name. Since it was repetitive, we have created a function to avoid code repetition. The advantage of defining this function inside another function is that, we can directly access local variables of `power_find_dirref` inside `checkStart` for data manipulation. This will improve code readability, since we don't need to pass local variable as arguments.